packages/runtime-dom/src/directives/vModel.ts

```diff
@@ -198,26 +204,26 @@ function callModelHook(
   binding: DirectiveBinding,
   vnode: VNode,
   prevVNode: VNode | null,
-  hook: keyof Directive
+  hook: keyof ObjectDirective
 ) {
-  let modelToUse: Directive
+  let modelToUse: ObjectDirective
   switch (el.tagName) {
     case 'SELECT':
-      modelToUse = vModelSelect
+      modelToUse = vModelSelect as ObjectDirective
```

Comment:
I meant the previous change was good in that we should revert to `const vModelSelect: ObjectDirective<...> = ...` so we can avoid the casting here.